.github/workflows/build.yml

```diff
@@ -24,12 +55,98 @@ jobs:
           python-version: "3.8"
       - uses: pre-commit/action@v2.0.0
 
-  build:
+  build-windows:
+    needs:
+      - init
     strategy:
+      matrix:
+        os: ${{ fromJSON(needs.init.outputs.build_windows) }}
+        python-version: ${{ fromJSON(needs.init.outputs.python_versions) }}
       fail-fast: false
+    defaults:
+      run:
+        shell: bash
+    runs-on: ${{ matrix.os }}
+
+    env:
+      BOOST_DIR: 3rdparty/boost
+      BOOST_VERSION: "1.76.0"
+      SKDECIDE_SKIP_DEPS: 1
+
+    steps:
+      - name: Checkout scikit-decide source code
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+          fetch-depth: 0
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Load cached venv
+        id: cached-pip-wheels
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache
+          key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Restore Boost cache
+        uses: actions/cache@v2.1.4
+        id: cache-boost
+        with:
+          path: ${{env.BOOST_DIR}}
+          key: BOOST_${{env.BOOST_VERSION}}
+
+      - name: Install Boost
+        if: steps.cache-boost.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p $BOOST_DIR
+          curl --silent --location --output - \
+            https://boostorg.jfrog.io/artifactory/main/release/$BOOST_VERSION/source/boost_${BOOST_VERSION//./_}.tar.bz2 |\
+            tar jxf - -C $BOOST_DIR --strip-components=1 boost_${BOOST_VERSION//./_}/boost
+        shell: bash
+
+      - name: Restore build dependencies
+        id: cache-build-dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            skdecide/hub/bin
+            skdecide/hub/share
+            skdecide/hub/*.msc
+          key: ${{ runner.os }}-cache-deps-${{ hashFiles('cpp/deps/chuffed', 'cpp/deps/gecode', 'cpp/deps/libminizinc') }}
+
+      - name: Update SKDECIDE_SKIP_DEPS
+        if: steps.cache-build-dependencies.outputs.cache-hit != 'true'
+        run: echo "SKDECIDE_SKIP_DEPS=0" >> $GITHUB_ENV
+
+      - name: Build Windows wheel
```

Comment:
Remove mentions of OS from step names (in all steps), this is no more useful when jobs are split by OS.

Reply:
done

Reply:
```sh
$ grep -i -E 'name:.*(linux|macos|windows)' .github/workflows/build.yml
      - name: Install omp on MacOS
      - name: Install and restore ccache for macOs
      - name: Let cmake use ccache for macOs
      - name: Restore docker dev image for Linux
      - name: Restore ccache cache for Linux
```

Reply:
> grep -i -E 'name:.*(linux|macos|windows)' .github/workflows/build.yml                                                [🐍 scikit-decide]
>

Reply:
fixed